Remotion/Web/Development.WebTesting/ScreenshotCreation/ScreenshotTransformationContext.cs

```diff
@@ -25,31 +25,77 @@ namespace Remotion.Web.Development.WebTesting.ScreenshotCreation
   /// <summary>
   /// A context for <see cref="IScreenshotTransformation{T}"/> to apply their transformations on.
   /// </summary>
-  public class ScreenshotTransformationContext<T>
+  public class ScreenshotTransformationContext<T> : IDisposable
   {
+    public static ScreenshotTransformationContext<T> CreateForTransformation (
+        ScreenshotManipulation manipulation,
+        [NotNull] Graphics graphics,
+        [NotNull] IScreenshotElementResolver<T> resolver,
+        [NotNull] T target,
+        CoordinateSystem coordinateSystem,
+        [NotNull] IScreenshotTransformation<T> transformation,
+        [NotNull] IBrowserContentLocator locator)
+    {
+      ArgumentUtility.CheckNotNull ("graphics", graphics);
+      ArgumentUtility.CheckNotNull ("resolver", resolver);
+      ArgumentUtility.CheckNotNull ("target", target);
+      ArgumentUtility.CheckNotNull ("transformation", transformation);
+      ArgumentUtility.CheckNotNull ("locator", locator);
+
+      var resolvedElement = Resolve (resolver, target, locator, coordinateSystem);
+      var context = new ScreenshotTransformationContext<T> (manipulation, graphics, resolver, target, resolvedElement, transformation);
+
+      context = transformation.BeginApply (context);
+
+      var parentBounds = context.ResolvedElement.ParentBounds;
+      if (parentBounds.HasValue)
+      {
+        return
+            context.CloneWith (
+                resolvedElement:
+                resolvedElement.CloneWith (elementBounds: Rectangle.Intersect (parentBounds.Value, context.ResolvedElement.ElementBounds)));
+      }
+      else
+      {
+        return context;
+      }
+    }
+
     private readonly ScreenshotManipulation _manipulation;
     private readonly Graphics _graphics;
     private readonly IScreenshotElementResolver<T> _resolver;
     private readonly T _target;
     private readonly ResolvedScreenshotElement _resolvedElement;
+    private readonly IScreenshotTransformation<T> _transformation;
 
     public ScreenshotTransformationContext (
         ScreenshotManipulation manipulation,
         [NotNull] Graphics graphics,
         [NotNull] IScreenshotElementResolver<T> resolver,
         [NotNull] T target,
         [NotNull] ResolvedScreenshotElement resolvedElement)
+        : this (manipulation, graphics, resolver, target, resolvedElement, null)

```

Comment:
Does it make sense to initialize a `ScreenshotTransformationContext` without a `ScreenshotTransformation`? Could we perhaps only leave the ctor that requires a transformation and leave out this ctor and the factory method?